### PR TITLE
[TemplateLiteral] Basic lexer implementation

### PR DESCRIPTION
**Pull request by croraf**
Very basic template literal implementation, of lexer only.

### CODE REVIEW

File: boa/src/syntax/lexer/tests.rs [hunk 1]
```diff
@@ -37,6 +37,29 @@ fn check_string() {
     assert_eq!(lexer.tokens[1].kind, TokenKind::string_literal("bbb"));
 }
 
+#[test]
+fn check_template_literal_simple() {
+    let s = "`I'm a template literal`";
```
Comment (jasonwilliams) [inline boa/src/syntax/lexer/tests.rs:42]: You shouldn’t need double quotes here

File: boa/src/syntax/lexer/tests.rs [hunk 2]
```diff
@@ -37,6 +37,29 @@ fn check_string() {
...
+    lexer.lex().expect("failed to lex");
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::template_literal("I'm a template literal")
+    );
+}
+
+#[test]
+fn check_template_literal_unterminated() {
+    let s = "`I'm a template";
```
Comment (jasonwilliams) [inline boa/src/syntax/lexer/tests.rs:53]: Same here

File: boa/src/syntax/ast/token.rs [hunk 3]
```diff
@@ -239,6 +239,8 @@ pub enum TokenKind {
     /// A string literal.
     StringLiteral(Box<str>),
 
+    TemplateLiteral(Box<str>),
```
Comment (Razican) [inline boa/src/syntax/ast/token.rs:242]: This would need some documentation. Ideally also a link to the spec.

Comment (croraf) [reply boa/src/syntax/ast/token.rs:242]: Not sure what you mean.

Comment (Razican) [reply boa/src/syntax/ast/token.rs:242]: At least, the variant should be documented like this:
```suggestion
    /// A template literal.
    TemplateLiteral(Box<str>),
```

Comment (croraf) [reply boa/src/syntax/ast/token.rs:242]: Commenting a variable named TemplateLiteral with a commend saying "template literal" is bad practice. I guess the comment with the possible link to the spec can go in boa/src/syntax/lexer/mod.rs. I put it there if you check that file in this PR.

**Pull request merged**